### PR TITLE
Add a silhouette option to reference images

### PR DESCRIPTION
**Pull request by TheLsbt**
This pull request adds a option to make images a silhouette. <br>
This is done by darkening all the opaque pixels. <br>

This is the UI changes: <br>
Screenshot (40)

### CODE REVIEW

**Comment by TheLsbt** [comment]
I think I fixed the linter issue.

File: src/UI/ReferenceImages/ReferenceImage.gd [hunk 1]
```diff
@@ -63,6 +65,14 @@ func deserialize(d: Dictionary) -> void:
 			# don't do FLAG_REPEAT - it could cause visual issues
 			itex.create_from_image(img, Texture.FLAG_MIPMAPS)
 			texture = itex
+		# Apply the silhouette shader
+		var mat = ShaderMaterial.new()
+		mat.shader = preload("res://src/Shaders/SilhouetteShader.gdshader")
```
Comment (OverloadedOrama) [inline src/UI/ReferenceImages/ReferenceImage.gd:70]: It's generally better to preload stuff in the variable declaration space and then use that variable later, for example by creating a variable called shader and setting it to this `var shader := preload("res://src/Shaders/SilhouetteShader.gdshader")` at the top of the code and then in this line you could do `mat.shader = shader`.

File: src/Preferences/PreferencesDialog.tscn [hunk 2]
```diff
@@ -11,14 +11,14 @@
 [sub_resource type="ButtonGroup" id=1]
 
 [node name="PreferencesDialog" type="AcceptDialog"]
-margin_right = 620.0
-margin_bottom = 450.0
+visible = true
```
Comment (OverloadedOrama) [inline src/Preferences/PreferencesDialog.tscn:14]: Looks like you might've forgotten to set it back to invisible 😅. From what I see, the changes in PreferencesDialog.tscn are just margin value changes that Godot makes automatically and don't really need to be included, unless you've manually made some changes to the nodes yourself. If you haven't, it's safe to just discard the changes made in this file completely.

**Comment by TheLsbt** [comment]
I believe that the issues with the Preference Dialog should be fixed (The file doesn't appear in **Files Changed**).  
I also created a variable to assign the shader to in ```RefrenceImage.gd```.

**Review by OverloadedOrama — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**